### ISSUE TITLE
Allow access to some more photos 

### ISSUE THREAD

**Issue by idevChandra6** [hyperoslo/Gallery#195]
With the new app privacy settings, we can allow some photos accessed by the application. Also, we can trigger to allow user to select some more photos accessed by the app.
Something like "Select more photos..." shown in this image https://i.stack.imgur.com/VG7p9.jpg. 

The app does not trigger this alert always. We can manually trigger this. Wondering if you can add this method call through a button on the picker. Please let us know.

Thank you.

**Comment by Rj707** [comment]
Plus the library doesn't refresh the more photos selected, you will have to either kill the app or go back and come on the screen where you are using Gallery library to view the recently more selected photos.

**Comment by idevchandra** [comment]
> Plus the library doesn't refresh the more photos selected, you will have to either kill the app or go back and come on the screen where you are using Gallery library to view the recently more selected photos.

Yes, I end up doing the same. I really wish "Add More" feature would be implemented.

Attached the screenshot of Reddit iOS app.
![IMG_C5F7E3D3E745-1](https://user-images.githubusercontent.com/61815678/154910280-fae89970-33aa-4238-92ab-26efc67cf306.jpeg)

**Comment by Rj707** [comment]
in my case when the user lands on the Screen where the **Gallery** is being used, for the first time upon selecting photos(**Select Photos**) the photos being displayed by **Gallery** are refreshed. But the next time, upon selecting more photos(**Select More Photos**) the photos being displayed by **Gallery** are  **NOT** refreshed. 
![Select](https://user-images.githubusercontent.com/9985642/154942200-c25f14f2-eb80-4c7f-8583-cb6022565aa8.png)
![More](https://user-images.githubusercontent.com/9985642/154942241-836e9faf-d5c4-402e-b7bf-b32bc8d07960.png)

**Comment by chandrapol** [comment]
> in my case when used lands on the Screen where the **Gallery** is being used, for the first time upon selecting photos(**Select Photos**) the photos being displayed by **Gallery** are refreshed. But the next time, upon selecting more photos(**Select More Photos**) the photos being displayed by **Gallery** are **NOT** refreshed.

I have this behavior too! When more photos are selected, the images are not refreshed.
Looks like this repo is not managed anymore.

**Comment by Rj707** [comment]
the only solution to this is set the **PHPhotoLibraryPreventAutomaticLimitedAccessAlert** set to **true** in info.plist

**Comment by Rj707** [comment]
found another quick fix to this, though not a good one. 

`extension GalleryController: PHPhotoLibraryChangeObserver {
    
    public func photoLibraryDidChange(_ changeInstance: PHChange) {
        DispatchQueue.main.async { [unowned self] in
            // Obtain authorization status and update UI accordingly
            setup()

            if let pagesController = makePagesController() {
              g_addChildController(pagesController)
            } else {
              let permissionController = makePermissionController()
              g_addChildController(permissionController)
            }
        }
    }
}`
